src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -69,9 +69,9 @@ public string UserProfileDir
 
         public Uri SwaggerEndpoint { get; set; }
 
-        public HttpState()
+        public HttpState(HttpClient httpClient = null)
         {
-            Client = new HttpClient();
+            Client = (httpClient == null)? new HttpClient() : httpClient;
```

Comment:
The constructor should force the dependency to be provided.

Reply:
Updated here:
https://github.com/aspnet/HttpRepl/commit/637d46558c13a50ae6a569c01c0dd404e16c42a6